### PR TITLE
RA-1610: <encounterDiagnoses*/> to specify a 'diagnosisConceptSources' attribute.

### PR DESCRIPTION
**Pull request by Ruhanga**
Ticket : https://issues.openmrs.org/browse/RA-1610

### CODE REVIEW

File: omod/src/main/java/org/openmrs/module/coreapps/fragment/controller/DiagnosesFragmentController.java [hunk 1]
```diff
@@ -85,24 +100,17 @@
...
-        Locale locale = context.getLocale();
-
-        List<ConceptSource> sources = emrApiProperties.getConceptSourcesForDiagnosisSearch();
-
-        List<ConceptSearchResult> hits = emrConceptService.conceptSearch(query, locale, null, diagnosisSets, sources, null);
-        List<SimpleObject> ret = new ArrayList<SimpleObject>();
-        for (ConceptSearchResult hit : hits) {
-            ret.add(simplify(hit, ui, locale));
-        }
+        List<SimpleObject> ret = search(context, ui, emrApiProperties, emrConceptService, conceptService, query, null, null, start, size);
```
Comment (mks-d) [inline omod/src/main/java/org/openmrs/module/coreapps/fragment/controller/DiagnosesFragmentController.java]: Ah thanks for this. I would return directly:
```java
return search(context, ui, emrApiProperties, emrConceptService, conceptService, query, null, null, start, size);
```
So we leave absolutely zero logic in the deprecated function.

File: omod/src/test/java/org/openmrs/module/coreapps/fragment/controller/DiagnosesFragmentControllerTest.java [hunk 2]
```diff
@@ -71,10 +71,37 @@ public void search_shouldSearchForDiagnosisConceptsUsingGloballyDefinedSuperSetG
...
 
         // verify
         verify(emrApiProperties, times(1)).getDiagnosisSets();
 
     }
+
+    @Test
+    public void search_shouldSearchForDiagnosesFromSpecifiedDiagnosisConceptSources() throws Exception {
+        // replay
+        controller.search(context, uiUtils, emrApiProperties, emrConceptService, conceptService, queryString, null, "ICRC,ICD-10-WHO", null, null);
```
Comment (mks-d) [inline omod/src/test/java/org/openmrs/module/coreapps/fragment/controller/DiagnosesFragmentControllerTest.java]: `"CIEL,ICD-10-WHO"`

File: omod/src/main/java/org/openmrs/module/coreapps/fragment/controller/DiagnosesFragmentController.java (hunk 1 above)
Comment (Ruhanga) [reply omod/src/main/java/org/openmrs/module/coreapps/fragment/controller/DiagnosesFragmentController.java]: Correct.

File: omod/src/test/java/org/openmrs/module/coreapps/fragment/controller/DiagnosesFragmentControllerTest.java (hunk 2 above)
Comment (Ruhanga) [reply omod/src/test/java/org/openmrs/module/coreapps/fragment/controller/DiagnosesFragmentControllerTest.java]: True this. Thank you.

**Pull request merged**